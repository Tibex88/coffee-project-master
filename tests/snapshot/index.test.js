/**
 * @jest-environment jsdom
 */

const {
  renderCoffee,
  renderCoffees,
  updateCoffees,
  createCoffee,
} = require("../../main");

const mockDocument = {
  querySelector: jest.fn(),
};

test("renderCoffee renders correctly", () => {
  const coffee = { id: 1, name: "Light City", roast: "light" };

  const html = renderCoffee(coffee);

  expect(html).toMatchSnapshot();
});

test("renderCoffees renders correctly", () => {
  const coffees = [
    { id: 1, name: "Light City", roast: "light" },
    { id: 2, name: "Half City", roast: "light" },
    { id: 2, name: "Half City", roast: "medium" },
  ];

  const html = renderCoffees(coffees);

  expect(html).toMatchSnapshot();
});

// test("updateCoffees renders correctly", () => {
//   const filteredCoffees = [
//     { id: 1, name: "Light City", roast: "light" },
//     { id: 2, name: "Half City", roast: "light" },
//   ];

//   const html = updateCoffees(filteredCoffees);

//   expect(html).toMatchSnapshot();
// });

// test("createCoffee renders correctly", () => {
//   const e = { preventDefault: jest.fn() };

//   const html = createCoffee(e);

//   expect(html).toMatchSnapshot();
// });
