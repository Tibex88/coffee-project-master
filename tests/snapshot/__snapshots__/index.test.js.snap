// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderCoffee renders correctly 1`] = `"<div class="coffee"><div class=flex><div class="d-none">1</div><div class="name"><p>Light City</p></div><p class="roast">light</p></div><div class="del">X</div></div>"`;

exports[`renderCoffees renders correctly 1`] = `"<div class="coffee"><div class=flex><div class="d-none">1</div><div class="name"><p>Light City</p></div><p class="roast">light</p></div><div class="del">X</div></div><div class="coffee"><div class=flex><div class="d-none">2</div><div class="name"><p>Half City</p></div><p class="roast">light</p></div><div class="del">X</div></div><div class="coffee"><div class=flex><div class="d-none">2</div><div class="name"><p>Half City</p></div><p class="roast">medium</p></div><div class="del">X</div></div>"`;
